UnityAds/src/main/java/com/mopub/mobileads/UnityInterstitial.java

```diff
@@ -45,22 +47,45 @@ protected void load(@NonNull final Context context, @NonNull final AdData adData
         mContext = context;
 
         setAutomaticImpressionAndClickTracking(false);
-        UnityAds.load(mPlacementId);
 
         mUnityAdsAdapterConfiguration.setCachedInitializationParameters(context, extras);
 
         UnityRouter.getInterstitialRouter().addListener(mPlacementId, this);
         UnityRouter.getInterstitialRouter().setCurrentPlacementId(mPlacementId);
+
         initializeUnityAdsSdk(extras);
+
+        UnityAds.load(mPlacementId, new IUnityAdsLoadListener() {
+            @Override
+            public void onUnityAdsAdLoaded(String placementId) {
+                if (mLoadListener != null) {
+                    mLoadListener.onAdLoaded();
+                    MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
+                }
+            }
+
+            @Override
+            public void onUnityAdsFailedToLoad(String placementId) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity interstitial failed to load for placement " + placementId);
+
+                if (mInteractionListener != null) {
+                    mInteractionListener.onAdFailed(MoPubErrorCode.NETWORK_NO_FILL);
+                }
+                MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, MoPubErrorCode.NO_FILL.getIntCode(), MoPubErrorCode.NETWORK_NO_FILL);
+            }
+        });
+
+        mUnityAdsAdapterConfiguration.setCachedInitializationParameters(context, extras);
```

Comment:
this might have been a merge issue.  I didn't add this line on my own...

Reply:
@UnitySteven  my bad, i saw in our previous adapter code, this line was called after load call, I meant to delete this line 51 and added here